lib/ember-cli/engine.rb

```diff
@@ -1,12 +1,27 @@
 module EmberCli
   class Engine < Rails::Engine
+    initializer "ember-cli-rails.rendering" do
+      require "ember-cli/ember_controller"
+      require "ember-cli/route_helpers"
+    end
+
     initializer "ember-cli-rails.enable" do
       EmberCli.enable! unless EmberCli.skip?
     end
 
-    initializer "ember-cli-rails.helpers" do
-      config.to_prepare do
-        ActionController::Base.helper EmberRailsHelper
+    config.to_prepare do
+      ActionController::Base.helper EmberRailsHelper
+    end
+
+    config.after_initialize do
+      if defined?(ApplicationController)
+        ApplicationController.class_eval do
+          before_action do
```

Comment:
Would it make sense for this to live in a module that you `ApplicationController.extend(module_name)` here?